spec/rspec/mocks/test_double_spec.rb

```diff
@@ -48,6 +48,17 @@ module Mocks
           end
         end
       end
+
+      pending "test doubles able to register themselves with ObjectInspector" do
+        allow_any_instance_of(TestDouble).to receive(:name).and_return("FooBar")
+        double('User')
+
+        RSpec::Support::ObjectInspector.register TestDouble do |user|
+          "User (#{user.name})"
+        end
+
+        expect(double.inspect).to eq("User (FooBar)")
```

Comment:
> This is still wrong for a spec, as it actually implements it, so the spec should just be:

> ```Ruby
it 'registers itself with the ObjectInspector' do
  expect(double('FooBar').inspect).to eq 'Double (FooBar)'
end
```

That's not right.  We're not touching the `double.inspect` method.  We're registering the type with `ObjectInspector` so that `ObjectInspector.inspect(double)` can be used to format the double in failure messages.  The spec should likewise use `ObjectInspector.inspect`.

> Although I'd argue it'd be nicer if the `ObjectInspector` could be registered with instances, then we could register specific instances when creating the doubles.

I disagree.  It would cause a memory leak.  If we registered instances, then Ruby's GC would not garbage collect those instances since there would still be a live references to then.  In addition, the `ObjectInspector` algorithm does a linear scan over each of the registered classes, and if we used instances, the number of objects to check against would grow and it would get slower.

We could of course deregister it but that seems much more complicated than just registering the class.  In addition, the object inspector logic up to this point has really been a function of what class/type the object is -- hashes are formatted one way, `Time` objects another way, etc.